packages/oasis-actions/src/strategies/aave/open/open.ts

```diff
@@ -0,0 +1,387 @@
+import BigNumber from 'bignumber.js'
+import { providers } from 'ethers'
+
+import { amountFromWei, amountToWei } from '../../../helpers'
+import { IBaseSimulatedTransition, Position } from '../../../helpers/calculations/Position'
+import { RiskRatio } from '../../../helpers/calculations/RiskRatio'
+import { DEFAULT_FEE, TYPICAL_PRECISION, ZERO } from '../../../helpers/constants'
+import { acceptedFeeToken } from '../../../helpers/swap/acceptedFeeToken'
+import { getSwapDataHelper } from '../../../helpers/swap/getSwapData'
+import * as operations from '../../../operations'
+import { AAVEStrategyAddresses } from '../../../operations/aave/v2'
+import { AAVEV3StrategyAddresses } from '../../../operations/aave/v3'
+import { AaveProtocolData, AaveProtocolDataArgs } from '../../../protocols/aave/getAaveProtocolData'
+import { Address, IOperation, IPositionTransition, PositionType, SwapData } from '../../../types'
+import { AAVETokens } from '../../../types/aave'
+import { getAaveTokenAddresses } from '../getAaveTokenAddresses'
+import { getCurrentPosition } from '../getCurrentPosition'
+
+interface AaveOpenArgs {
+  depositedByUser?: {
+    collateralToken?: { amountInBaseUnit: BigNumber }
+    debtToken?: { amountInBaseUnit: BigNumber }
+  }
+  multiple: BigNumber
+  slippage: BigNumber
+  positionType: PositionType
+  collateralToken: { symbol: AAVETokens; precision?: number }
+  debtToken: { symbol: AAVETokens; precision?: number }
+}
+
+interface AaveOpenDependencies {
+  addresses: AAVEStrategyAddresses | AAVEV3StrategyAddresses
+  proxy: Address
+  user: Address
+  isDPMProxy: boolean
+  /* Services below 👇*/
+  provider: providers.Provider
+  protocol: {
+    version: 2 | 3
```

Comment:
like before, make an enum. On the frontend we have:
https://github.com/OasisDEX/oasis-borrow/blob/dev/lendingProtocols/LendingProtocol.ts

Reply:
DOne